### PR TITLE
test: cache all JSON-LD contexts in DID tests

### PR DESCRIPTION
**Pull request by kdimak**
Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

#1846

### CODE REVIEW

File: pkg/doc/did/doc.go [hunk 1]
```diff
@@ -762,7 +760,7 @@ func (doc *Doc) JSONBytes() ([]byte, error) {
 }
 
 // VerifyProof verifies document proofs
-func (doc *Doc) VerifyProof(suites ...verifier.SignatureSuite) error {
+func (doc *Doc) VerifyProof(suites []verifier.SignatureSuite, jsonldOpts ...jsonld.ProcessorOpts) error {
```
Comment (kdimak) [inline pkg/doc/did/doc.go:763]: Here we pass universal `jsonld.ProcessorOpts`, one of that is `WithDocumentLoader()`. The configured JSON-LD document loader with cached contexts is used in the tests, but one can pass arbitrary document loader in production code (with e.g. custom setup of HTTP client).

File: pkg/doc/did/contexts.go [hunk 2]
```diff
@@ -1,294 +0,0 @@
-/*
```
Comment (llorllale) [inline pkg/doc/did/contexts.go:1]: why was this deleted and then moved to .jsonld files? Don't we need these contexts in production?

Comment (kdimak) [reply pkg/doc/did/contexts.go:1]: Right, it was moved to the `*.jsonld` files of test data directory. 
I have no strong opinion on whether we should have the cached contexts (by default) in the production.
What do you think @troyronda ?

Comment (llorllale) [reply pkg/doc/did/contexts.go:1]: I think the default should be cached. The question now is where to maintain them, in files or in go code....

Go code is more straightforward imo.

Comment (fqutishat) [reply pkg/doc/did/contexts.go]: I agree we need them in Go code

Comment (kdimak) [reply pkg/doc/did/contexts.go]: Yes, this makes sense. I added back the caching to the production code. It's also exported now: `CachingJSONLDLoader()`. This allows the users to use the default caching or apply their own via providing custom loader as `jsonld.ProcessorOpts`.
